waiter/src/waiter/scheduler.clj

```diff
@@ -108,7 +108,17 @@
     "Retrieves the state the scheduler is maintaining for the given service-id.")
 
   (state [this]
-    "Returns the global (i.e. non-service-specific) state the scheduler is maintaining"))
+    "Returns the global (i.e. non-service-specific) state the scheduler is maintaining")
+
+  (validate-user [this ^String user ^String service-id]
+    "Verify that a user has the proper permissions to create services on the underlying scheduler platform."))
```

Comment:
Docstring should clearly state that an exception will be thrown if validation fails.